platform/android/app/src/main/kotlin/coop/polypoly/polypod/endpoint/Endpoint.kt

```diff
@@ -37,6 +37,43 @@ class Endpoint(
         return endpointsJson[endpointId]
     }
 
+    suspend fun uploadToServer(
+        context: Context,
+        endpointId: String,
+        errorMsg: String
+    ) {
+        if (errorMsg == null) {
```

Comment:
Nit: But can message be ever `null`? I may be wrong, but if `errorMsg` can be `null`, shouldn't it be specified as `String?` in function signature?